src/Test/Perf/helloworld/HelloWorld.csx

```diff
@@ -0,0 +1,20 @@
+#load "../util.csx"
+using System.IO;
+
+var pathToCsc = Path.Combine(BinDebugDirectory(), "csc.exe");
+var pathToHelloWorld = Path.Combine(MyWorkingDirectory(), "HelloWorld.cs");
+var pathToOutput = Path.Combine(MyArtifactsDirectory(), "HelloWorld.exe");
+
+ProcessResult result;
+
+var msToCompile = WalltimeMs(out result,
```

Comment:
I like the idea of timing even when "-noprofile" is passed, but I do worry about having tests programmatically opt-in to Timing/profiling (this is a potential gotcha that is somewhat difficult to audit once you have more than a few tests).  It would be nice if Iterate did this by default.  I imagine having a file on the side or some sort of metadata associated with the test that declares a set of process names and metrics we want to collect during the test.  I imagine this file would be the same across all compiler tests, across all IDE tests, etc.  If there are scenarios where we really don't want to collect this data (like checking binary sizes), we could have an empty file (that doesn't specify any metrics).